adapta/security/clients/aws/_aws_client.py

```diff
@@ -80,15 +84,33 @@ def connect_account(self):
         """
 
     def get_pyarrow_filesystem(self, path: DataPath, connection_options: Optional[Dict[str, str]] = None) -> FileSystem:
-        raise ValueError("Not supported  in AwsClient")
+        """
+        Not supported in AwsClient.
+        :return:
+        """
 
-    def initialize_session(self) -> "AwsClient":
+    def initialize_session(self, session_callable=None) -> "AwsClient":
         """
-        Initializes session. Should be called before any operations with client
+        Initializes the session by custom session function or a default one if no function is provided."
+        :return: AwsClient with established session.
         """
-        self._session = boto3.Session(
+        if session_callable is None:
```

Comment:
Should skip if session is initialised?

Reply:
That's a good idea. I've added it now in [3943e2d](https://github.com/SneaksAndData/adapta/pull/437/commits/3943e2dda0291afba169f072092af8a0ff6741b7)
Thank you for the suggestion 🙂